utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -58,6 +58,14 @@ def is_build_script_impl_product(cls):
         """
         raise NotImplementedError
 
+    @classmethod
+    def is_before_build_script_impl_product(cls):
+        """is_before_build_script_impl_product -> bool
+
+        Whether this product is build before any build-script-impl products.
```

Comment:
I would add to the doc here that there is an assumption that this is a non-build-script-impl product.

Also, it may be worth it to use a NotImplementedError and then to override it in sub-classes if there is a nice place to do it. That way if someone forgets they get an error. That being said, I am not completely wedded to that.

Reply:
Done, and added `NotImplementedError` here.

Reply:
Looks good!